Description: Restore ability to create launcher
 Restores the ability to create a .desktop launcher (shortcut) with the
 nautilus context menu
 .
 nautilus (3.4.3solusos1) two; urgency=low
 .
   * New upstream release.
   * Refresh debian/patches/12_list-view_expand.patch.
Author: Ikey Doherty <ikey@solusos.com>

---
Reviewed-By: Ikey Doherty <ikey@solusos.com>
Last-Update: <2012-06-06>

--- nautilus-3.4.2.orig/src/nautilus-desktop-icon-view-ui.xml
+++ nautilus-3.4.2/src/nautilus-desktop-icon-view-ui.xml
@@ -4,6 +4,7 @@
                 <placeholder name="New Window Items">
                 </placeholder>
                 <placeholder name="New Object Items">
+                        <menuitem name="New Launcher" action="New Launcher Desktop"/>
                 </placeholder>
         </placeholder>
         <placeholder name="After Zoom Items">
--- nautilus-3.4.2.orig/src/nautilus-actions.h
+++ nautilus-3.4.2/src/nautilus-actions.h
@@ -114,6 +114,8 @@
 #define NAUTILUS_ACTION_LOCATION_STOP_VOLUME "Location Stop Volume"
 #define NAUTILUS_ACTION_LOCATION_POLL "Location Poll"
 #define NAUTILUS_ACTION_SCRIPTS "Scripts"
+#define NAUTILUS_ACTION_NEW_LAUNCHER "New Launcher"
+#define NAUTILUS_ACTION_NEW_LAUNCHER_DESKTOP "New Launcher"
 #define NAUTILUS_ACTION_NEW_DOCUMENTS "New Documents"
 #define NAUTILUS_ACTION_NEW_EMPTY_DOCUMENT "New Empty Document"
 #define NAUTILUS_ACTION_EMPTY_TRASH_CONDITIONAL "Empty Trash Conditional"
--- nautilus-3.4.2.orig/src/nautilus-directory-view-ui.xml
+++ nautilus-3.4.2/src/nautilus-directory-view-ui.xml
@@ -98,6 +98,7 @@
 	<placeholder name="Before Zoom Items">
 		<placeholder name="New Object Items">
 			<menuitem name="New Folder" action="New Folder"/>
+			<menuitem name="New Launcher" action="New Launcher"/>
 			<menu action="New Documents">
 		         	<menuitem name="No Templates" action="No Templates"/>
 			 	<placeholder name="New Documents Placeholder"/>
--- nautilus-3.4.2.orig/src/nautilus-desktop-icon-view.c
+++ nautilus-3.4.2/src/nautilus-desktop-icon-view.c
@@ -666,6 +666,23 @@ action_empty_trash_conditional_callback
 	nautilus_file_operations_empty_trash (GTK_WIDGET (data));
 }
 
+static void
+action_new_launcher_callback (GtkAction *action, gpointer data)
+{
+	char *desktop_directory;
+
+        g_assert (NAUTILUS_IS_VIEW (data));
+
+	desktop_directory = nautilus_get_desktop_directory ();
+
+	nautilus_launch_application_from_command (gtk_widget_get_screen (GTK_WIDGET (data)),
+						  "gnome-desktop-item-edit", 
+						  FALSE,
+						  "--create-new", desktop_directory, NULL);
+	g_free (desktop_directory);
+
+}
+
 static gboolean
 trash_link_is_selection (NautilusView *view)
 {
@@ -709,6 +726,13 @@ real_update_menus (NautilusView *view)
 
 	desktop_view = NAUTILUS_DESKTOP_ICON_VIEW (view);
 
+
+	/* New Launcher */
+	action = gtk_action_group_get_action (desktop_view->details->desktop_action_group,
+					      NAUTILUS_ACTION_NEW_LAUNCHER_DESKTOP);
+	gtk_action_set_visible (action,
+				TRUE);
+
 	/* Empty Trash */
 	include_empty_trash = trash_link_is_selection (view);
 	action = gtk_action_group_get_action (desktop_view->details->desktop_action_group,
@@ -726,6 +750,13 @@ real_update_menus (NautilusView *view)
 
 static const GtkActionEntry desktop_view_entries[] = {
 	/* name, stock id */
+	{ "New Launcher Desktop", NULL,
+	  /* label, accelerator */
+	  N_("Create L_auncher..."), NULL,
+	  /* tooltip */
+	  N_("Create a new launcher"),
+	  G_CALLBACK (action_new_launcher_callback) },
+	/* name, stock id */
 	{ "Change Background", NULL,
 	  /* label, accelerator */
 	  N_("Change Desktop _Background"), NULL,
--- nautilus-3.4.2.orig/src/nautilus-view.c
+++ nautilus-3.4.2/src/nautilus-view.c
@@ -2035,6 +2035,29 @@ action_new_empty_file_callback (GtkActio
 }
 
 static void
+action_new_launcher_callback (GtkAction *action,
+			      gpointer callback_data)
+{
+	char *parent_uri;
+	NautilusView *view;
+	GtkWindow *window;
+
+	g_assert (NAUTILUS_IS_VIEW (callback_data));
+
+	view = NAUTILUS_VIEW (callback_data);
+
+	parent_uri = nautilus_view_get_backing_uri (view);
+
+	window = nautilus_view_get_containing_window (view);
+	nautilus_launch_application_from_command (gtk_widget_get_screen (GTK_WIDGET (view)),
+						  "gnome-desktop-item-edit", 
+						  FALSE,
+						  "--create-new", parent_uri, NULL);
+
+	g_free (parent_uri);
+}
+
+static void
 action_properties_callback (GtkAction *action,
 			    gpointer callback_data)
 {
@@ -6964,6 +6987,10 @@ static const GtkActionEntry directory_vi
   /* label, accelerator */       N_("_Empty Document"), NULL,
   /* tooltip */                  N_("Create a new empty document inside this folder"),
 				 G_CALLBACK (action_new_empty_file_callback) },
+  /* name, stock id */         { "New Launcher", NULL,
+  /* label, accelerator */       N_("Create L_auncher..."), NULL,
+  /* tooltip */                  N_("Create a new launcher"),
+                                 G_CALLBACK (action_new_launcher_callback) },
   /* name, stock id */         { "Open", NULL,
   /* label, accelerator */       N_("_Open"), "<control>o",
   /* tooltip */                  N_("Open the selected item in this window"),
